packages/ui-components/src/components/Message/Message.types.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export type MessageType = "info" | "warning" | "danger" | "error" | "success"
+
+export interface MessageProps {
+  /** Optional message title */
+  title?: string
+  /** Optional description to render as content. Alternatively, content can be passed as children. */
+  text?: string
+  /** Optional variant for message appearance e.g. 'success'. Default is 'info'. */
+  variant?: MessageType
+  /** Optional: If true, the message will have a 'close' button. Default is false. */
+  dismissible?: boolean
+  /** Optional: If true, the message will auto-dismiss after 10 seconds or the specified autoDismissTimeout. */
+  autoDismiss?: boolean
+  /** Optional timeout in milliseconds after which the message auto-dismisses. Default is 10000 (10 seconds). */
+  autoDismissTimeout?: number
+  /** Optional handler function called once the message is dismissed */
+  onDismiss?: () => void
```

Comment:
This results in the type being displayed in a user-unfriendly way in Storybook.

![image](https://github.com/user-attachments/assets/0c507f2f-f648-4caf-adc7-c9ff81dc0b10)

I’m not an expert in TypeScript yet, but could `React.MouseEventHandler` be useful in this case?

Reply:
Check e.g. the `Button` component. It has been moved to TS and also has click event handlers

Reply:
Thank You @andypf and @edda.
I think that `React.MouseEventHandler` would be incorrect in this context. 
In the `Button` component, it's tied to the specific mouse event `onClick`, so `React.MouseEventHandler<EventTarget>` is correct.
But here, the `onDismiss` function is a custom handler. It should just be a function that takes no arguments and returns nothing - `(() => void)`.

Reply:
Got it, let’s wait until Wowa is here, maybe he has an idea.